include/libtorrent/bitfield.hpp

```diff
@@ -201,6 +206,32 @@ namespace libtorrent {
 					bit <<= 1;
 				}
 			}
+			void add(int n)
+			{
+				TORRENT_ASSERT(buf);
+				if (n == 0)
+					return;
+
+				if (n < 0)
+					return sub(-n);
+
+				buf += (n / 32);
+				int const remaining_bits = (n % 32);
+				for (int i = 0; i < remaining_bits; ++i) inc();
```

Comment:
This loop can be avoided if change iterator implementation to store buffer start + position instead of current word + bitmask.